File: src/main/java/com/smarthome/backend/model/Workflow.java
package com.smarthome.backend.model;

import java.util.List;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.google.gson.annotations.SerializedName;

import lombok.Data;

/**
 * Repräsentiert einen Workflow mit allen Knoten und deren Konfigurationen.
 */
@Data
public class Workflow {
    private List<Node> nodes;
    
    @SerializedName("startNodeId")
    private String startNodeId;
    private transient Logger logger = LoggerFactory.getLogger(Workflow.class);

    public Node getTriggerNode(){
        for(Node node : nodes) {
            if(node.getType().equals("trigger")) {
                return node;
            }
        }
        logger.warn("No trigger node found");
        return null;
    }
}

File: src/main/node/com/smarthome/backend/model/Workflow.ts
import { logger } from "../logger.js";
import { Node } from "./Node.js";

export class Workflow {
  nodes?: Node[];
  startNodeId?: string;

  constructor(init?: Partial<Workflow>) {
    Object.assign(this, init);
  }

  getTriggerNode() {
    if (!this.nodes || this.nodes.length === 0) {
      logger.warn("No trigger node found");
      return null;
    }
    const triggerNode = this.nodes.find(node => node.type === "trigger") ?? null;
    if (!triggerNode) {
      logger.warn("No trigger node found");
    }
    return triggerNode;
  }
}
